python/plugins/processing/algs/qgis/RasterSampling.py

```diff
@@ -0,0 +1,200 @@
+        )
+
+        columnPrefix = QgsProcessingParameterString(
+            self.COLUMN_PREFIX,
+            self.tr('Output column prefix'), 'rvalue'
+        )
+        columnPrefix.setFlags(columnPrefix.flags() | QgsProcessingParameterDefinition.FlagAdvanced)
+        self.addParameter(columnPrefix)
+
+        self.addParameter(
+            QgsProcessingParameterFeatureSink(
+                self.OUTPUT,
+                self.tr('Sampled Points')
+            )
+        )
+
+    def processAlgorithm(self, parameters, context, feedback):
+
+        source = self.parameterAsSource(
+            parameters,
+            self.INPUT,
+            context
+        )
+
+        sampled_raster = self.parameterAsRasterLayer(
+            parameters,
+            self.RASTERCOPY,
+            context
+        )
+
+        columnPrefix = self.parameterAsString(
+            parameters,
+            self.COLUMN_PREFIX,
+            context
+        )
+
+        if source is None:
+            raise QgsProcessingException(self.invalidSourceError(parameters, self.INPUT))
+
+        source_fields = source.fields()
+        raster_fields = QgsFields()
+
+        # append field to vector as columnPrefix_bandCount
+        for b in range(sampled_raster.bandCount()):
+            raster_fields.append(QgsField(
+                columnPrefix + str('_{}'.format(b + 1)), QVariant.Double
+            )
+            )
+
+        # combine all the vector fields
+        out_fields = QgsProcessingUtils.combineFields(source_fields, raster_fields)
+
+        (sink, dest_id) = self.parameterAsSink(
+            parameters,
+            self.OUTPUT,
+            context,
+            out_fields,
+            source.wkbType(),
+            source.sourceCrs()
+        )
+
+        if sink is None:
+            raise QgsProcessingException(self.invalidSinkError(parameters, self.OUTPUT))
+
+        total = 100.0 / source.featureCount() if source.featureCount() else 0
+        features = source.getFeatures()
+
+        # create the coordinates transformation context
+        ct = QgsCoordinateTransform(source.sourceCrs(), sampled_raster.crs(), context.transformContext())
+
+        for n, i in enumerate(source.getFeatures()):
+
+            if i.geometry().isMultipart():
+                raise QgsProcessingException(self.tr('''Impossible to sample data
+                of a Multipart layer. Please use the Multipart to single part
+                algorithm to transform the layer.'''))
+
+            # get the feature geometry as point
+            point = i.geometry().asPoint()
+
+            # reproject to raster crs
+            try:
+                point = ct.transform(point)
+            except QgsCsException:
+                for b in range(sampled_raster.bandCount()):
+                    attrs.append(
+                        None
+                    )
+                feedback.reportError(self.tr('Could not reproject feature {} to raster CRS').format(i.id()))
```

Comment:
The problem here is that the loop will continue after this, so will sample the raster at the invalid point coordinate. You need to add in the `except` block:

    i.setAttributes(attrs)
    sink.addFeature(i, QgsFeatureSink.FastInsert)
    feedback.setProgress(int(n * total))
    continue

(or refactor slightly to ensure that the sampling doesn't occur when the QgsCsException is thrown)

Reply:
ok added the code in the except block.. 

I saw that in the meanwhile the new raster sampling method has been added. Should I use that one? Usage is the same right?